### PR TITLE
Add success resource/page with boilerplate copy

### PR DESCRIPTION
**Pull request by jayroh**
As a client, after the final step they should see a confirmation page
telling them that they are FINISHED! In addition, give them guidance on
what to expect next.

Also:

* Rescue and provide a NullStep for instances where a form object class
  can not be inferred from the controller path.

Trello:

https://trello.com/c/DVStptea/82-as-a-client-after-i-uploading-docs-or-choosing-not-to-i-want-to-see-a-confirmation-page-telling-me-i-am-finished-and-giving-guid

### CODE REVIEW

**Comment by jayroh** [comment]
[As a client, after I uploading docs or choosing not to, I want to see a confirmation page telling me I am FINISHED! and giving guidance on what to do next](https://trello.com/c/DVStptea/82-as-a-client-after-i-uploading-docs-or-choosing-not-to-i-want-to-see-a-confirmation-page-telling-me-i-am-finished-and-giving-guid)

File: app/controllers/send_application_controller.rb [hunk 1]
```diff
@@ -7,8 +7,7 @@ def update
     if @step.valid?
       current_snap_application.update(step_params)
       create_and_send_pdf
-      flash[:notice] = "You will receive an email with your filled out application attached in a few minutes."
-      redirect_to root_path(anchor: "fold")
+      redirect_to(next_path)
```
Comment (jessieay) [inline app/controllers/send_application_controller.rb]: no flash to let them know it was sent?

Comment (jessieay) [inline app/controllers/send_application_controller.rb]: Per Alan's comment in The Trello:

> If you email a copy to yourself, let's take you back to this page. We should give some confirmation that the email worked.

**Comment by jayroh** [comment]
Updated this to resemble the following (please excuse the weird offset/sizing):

michigan_assistance_programs
michigan_assistance_programs

File: app/views/documents/edit.html.erb [hunk 2]
```diff
@@ -44,7 +44,7 @@
           Add documents
         </button>
 
-        <%= link_to send_application_steps_path, class: "button" do %>
+        <%= link_to next_path, class: "button" do %>
```
Comment (jessieay) [inline app/views/documents/edit.html.erb]: 😸

File: spec/controllers/success_controller_spec.rb [hunk 3]
```diff
@@ -15,6 +15,14 @@
     end
   end
 
+  context "in order to not allow going back" do
```
Comment (jessieay) [inline spec/controllers/success_controller_spec.rb]: ✔️

**Comment by jayroh** [comment]
> Just to confirm: is the flash message actually showing up after an email is sent? I was having issues getting it to show up on a "step" page.

It is, yeah. Here it is on the homepage after the redirect:



And I think maybe the flash not showing up might have to do with these being set on `flash` instead of `flash.now`. I'll take a look and confirm

**Comment by jayroh** [comment]
Looks like the only place we're setting the flash is right before we redirect to the homepage. Should we be setting a message of some sort here?

1__tmux

In the else block?

**Comment by jessieay** [comment]
@jayroh did you mean to close without merge?

**Pull request closed without merging**